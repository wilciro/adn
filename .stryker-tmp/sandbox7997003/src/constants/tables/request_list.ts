// @ts-nocheck
function stryNS_9fa48() {
  var g = new Function("return this")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

import { formatPrice } from 'utils/text';
export const requestListHeader = stryMutAct_9fa48("205") ? [] : (stryCov_9fa48("205"), [stryMutAct_9fa48("206") ? {} : (stryCov_9fa48("206"), {
  label: stryMutAct_9fa48("207") ? "" : (stryCov_9fa48("207"), 'Mascota'),
  row: stryMutAct_9fa48("208") ? "" : (stryCov_9fa48("208"), 'pet_name')
}), stryMutAct_9fa48("209") ? {} : (stryCov_9fa48("209"), {
  label: stryMutAct_9fa48("210") ? "" : (stryCov_9fa48("210"), 'Tipo mascota'),
  row: stryMutAct_9fa48("211") ? "" : (stryCov_9fa48("211"), 'pet_type')
}), stryMutAct_9fa48("212") ? {} : (stryCov_9fa48("212"), {
  label: stryMutAct_9fa48("213") ? "" : (stryCov_9fa48("213"), 'Edad mascota'),
  row: stryMutAct_9fa48("214") ? "" : (stryCov_9fa48("214"), 'pet_age')
}), stryMutAct_9fa48("215") ? {} : (stryCov_9fa48("215"), {
  label: stryMutAct_9fa48("216") ? "" : (stryCov_9fa48("216"), 'Dueño'),
  row: stryMutAct_9fa48("217") ? "" : (stryCov_9fa48("217"), 'owner_name')
}), stryMutAct_9fa48("218") ? {} : (stryCov_9fa48("218"), {
  label: stryMutAct_9fa48("219") ? "" : (stryCov_9fa48("219"), 'Documento dueño'),
  row: stryMutAct_9fa48("220") ? "" : (stryCov_9fa48("220"), 'owner_document')
}), stryMutAct_9fa48("221") ? {} : (stryCov_9fa48("221"), {
  label: stryMutAct_9fa48("222") ? "" : (stryCov_9fa48("222"), 'Date'),
  row: stryMutAct_9fa48("223") ? "" : (stryCov_9fa48("223"), 'date')
}), stryMutAct_9fa48("224") ? {} : (stryCov_9fa48("224"), {
  label: stryMutAct_9fa48("225") ? "" : (stryCov_9fa48("225"), 'Precio'),
  row: stryMutAct_9fa48("226") ? "" : (stryCov_9fa48("226"), 'price'),
  format: stryMutAct_9fa48("227") ? () => undefined : (stryCov_9fa48("227"), (price: string) => formatPrice(price))
})]);