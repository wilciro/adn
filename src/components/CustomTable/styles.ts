import styled from 'styled-components';

export const TableResponsive = styled.div`
    width: 100%;
    overflow: auto;
`;
