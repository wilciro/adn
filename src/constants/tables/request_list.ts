import { formatPrice } from 'utils/text';

export const requestListHeader = [
  {
    label: 'Mascota',
    row: 'pet_name',
  },
  {
    label: 'Tipo mascota',
    row: 'pet_type',
  },
  {
    label: 'Edad mascota',
    row: 'pet_age',
  },
  {
    label: 'Dueño',
    row: 'owner_name',
  },
  {
    label: 'Documento dueño',
    row: 'owner_document',
  },
  {
    label: 'Fecha',
    row: 'date',
  },
  {
    label: 'Hora',
    row: 'time',
  },
  {
    label: 'Precio',
    row: 'price',
    format: (price: string) => formatPrice(price),
  },
];
