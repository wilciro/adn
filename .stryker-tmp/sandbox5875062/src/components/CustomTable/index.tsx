// @ts-nocheck
// React
function stryNS_9fa48() {
  var g = new Function("return this")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

import React, { useEffect, useState } from 'react'; // Mantine

import { Table } from '@mantine/core';
import { getTableData } from 'services/tableService';
import { TableResponsive } from './styles';
interface HeaderModel {
  label: string;
  row: string;
  format?: (row: string) => string;
}
export interface CustomTableProps {
  header: Array<HeaderModel>;
  title: string;
  endpoint: string;
  rows?: Array<object>;
}

const CustomTable: React.FC<CustomTableProps> = ({
  header,
  title,
  endpoint,
  rows = stryMutAct_9fa48("51") ? ["Stryker was here"] : (stryCov_9fa48("51"), [])
}) => {
  if (stryMutAct_9fa48("52")) {
    {}
  } else {
    stryCov_9fa48("52");
    const [rowsData, setRowsData] = useState<Array<object>>(rows);

    const getData = async () => {
      if (stryMutAct_9fa48("53")) {
        {}
      } else {
        stryCov_9fa48("53");

        if (stryMutAct_9fa48("57") ? endpoint.length <= 0 : stryMutAct_9fa48("56") ? endpoint.length >= 0 : stryMutAct_9fa48("55") ? false : stryMutAct_9fa48("54") ? true : (stryCov_9fa48("54", "55", "56", "57"), endpoint.length > 0)) {
          if (stryMutAct_9fa48("58")) {
            {}
          } else {
            stryCov_9fa48("58");
            const data = await getTableData(endpoint);

            if (stryMutAct_9fa48("60") ? false : stryMutAct_9fa48("59") ? true : (stryCov_9fa48("59", "60"), data.valid)) {
              if (stryMutAct_9fa48("61")) {
                {}
              } else {
                stryCov_9fa48("61");
                setRowsData(stryMutAct_9fa48("64") ? data.data && [] : stryMutAct_9fa48("63") ? false : stryMutAct_9fa48("62") ? true : (stryCov_9fa48("62", "63", "64"), data.data || (stryMutAct_9fa48("65") ? ["Stryker was here"] : (stryCov_9fa48("65"), []))));
              }
            }
          }
        }
      }
    };

    useEffect(() => {
      if (stryMutAct_9fa48("66")) {
        {}
      } else {
        stryCov_9fa48("66");
        getData();
      }
    }, stryMutAct_9fa48("67") ? ["Stryker was here"] : (stryCov_9fa48("67"), []));
    const ths = header.map(stryMutAct_9fa48("68") ? () => undefined : (stryCov_9fa48("68"), (item: HeaderModel, index: number) => <th key={stryMutAct_9fa48("69") ? `` : (stryCov_9fa48("69"), `header-${item.row}`)}>{item.label}</th>));
    return <TableResponsive>
      <Table striped highlightOnHover>
        <caption>{title}</caption>
        <thead>
          <tr>{ths}</tr>
        </thead>
        <tbody>
          {rowsData.map((row: object, index: number) => {
            if (stryMutAct_9fa48("70")) {
              {}
            } else {
              stryCov_9fa48("70");
              return <tr key={stryMutAct_9fa48("71") ? `` : (stryCov_9fa48("71"), `column-${index}`)}>
                {header.map((itemH: HeaderModel, iHead: number) => {
                  if (stryMutAct_9fa48("72")) {
                    {}
                  } else {
                    stryCov_9fa48("72");

                    if (stryMutAct_9fa48("74") ? false : stryMutAct_9fa48("73") ? true : (stryCov_9fa48("73", "74"), itemH.format)) {
                      if (stryMutAct_9fa48("75")) {
                        {}
                      } else {
                        stryCov_9fa48("75");
                        return <td key={stryMutAct_9fa48("76") ? `` : (stryCov_9fa48("76"), `${index}-${iHead}`)}>
                        {itemH.format(itemH.row in row ? row[(itemH.row as keyof typeof row)] : stryMutAct_9fa48("77") ? "Stryker was here!" : (stryCov_9fa48("77"), ''))}
                      </td>;
                      }
                    }

                    return <td key={stryMutAct_9fa48("78") ? `` : (stryCov_9fa48("78"), `${index}-${iHead}`)}>
                      {itemH.row in row ? row[(itemH.row as keyof typeof row)] : stryMutAct_9fa48("79") ? "Stryker was here!" : (stryCov_9fa48("79"), '')}
                    </td>;
                  }
                })}
              </tr>;
            }
          })}
        </tbody>
      </Table>
    </TableResponsive>;
  }
};

CustomTable.defaultProps = stryMutAct_9fa48("80") ? {} : (stryCov_9fa48("80"), {
  rows: stryMutAct_9fa48("81") ? ["Stryker was here"] : (stryCov_9fa48("81"), [])
});
export default CustomTable;