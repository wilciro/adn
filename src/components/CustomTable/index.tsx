// React
import React, { useEffect, useState } from 'react';

// Mantine
import { Table } from '@mantine/core';
import { getTableData } from 'services/tableService';
import { TableResponsive } from './styles';

interface HeaderModel {
    label: string,
    row: string,
    format?: (row: string) => string
}

export interface CustomTableProps {
    header: Array<HeaderModel>,
    title: string,
    endpoint: string
}

const CustomTable: React.FC<CustomTableProps> = ({header, title, endpoint}) => {

    const [rowsData, setRowsData] = useState<Array<object>>([]);
    
    const getData = async () => {
        const data = await getTableData(endpoint);
        if (data.valid) {
            setRowsData(data.data);
        }
    };
    
    useEffect(() => {
        getData();
    }, [])

    const ths = header.map((item: HeaderModel, index: number) => (
        <th key={`header-${item.row}`}>{item.label}</th>
    ));

    return (
        <TableResponsive>
            <Table striped highlightOnHover>
                <caption>{title}</caption>
                <thead>
                    <tr>
                        {ths}
                    </tr>
                </thead>
                <tbody>
                    {
                        rowsData.map((row: object, index: number) => {      
                            return (
                                <tr key={`column-${index}`}>
                                    {
                                        header.map((itemHeader: HeaderModel, indexHeader: number) => {
                                            if (itemHeader.format)  {
                                                return <td key={`row-${index}-${indexHeader}`}>{itemHeader.format(itemHeader.row in row ? row[itemHeader.row as keyof typeof row] : "")}</td>
                                            }
                                            return <td key={`row-${index}-${indexHeader}`}>{itemHeader.row in row ? row[itemHeader.row as keyof typeof row] : ""}</td>
                                        })
                                    }
                                </tr>
                            )
                        })
                    }    
                </tbody>
            </Table>
        </TableResponsive>
    );
};

export default CustomTable;

