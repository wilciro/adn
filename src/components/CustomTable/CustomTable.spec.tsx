import React from 'react';
import { render } from '@testing-library/react';
import { requestListHeader } from 'constants/tables/request_list';
import { TableResponsive } from './styles';
import CustomTable from '.';

describe('CustomTable tests', () => {
  it('should match style', () => {
    const { container } = render(<TableResponsive>Table</TableResponsive>);
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot CustomTable', () => {
    const { container } = render(
      <CustomTable
        header={requestListHeader}
        title="Hola mundo"
        endpoint="requests"
      />,
    );

    expect(container.getElementsByTagName('caption')).toBeTruthy();
  });
});
