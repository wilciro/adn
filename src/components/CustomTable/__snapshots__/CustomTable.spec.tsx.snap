// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTable tests should match snapshot CustomTable 1`] = `
<div>
  <div
    class="sc-bczRLJ LwpKF"
  >
    <table
      class="mantine-Table-root __mantine-ref-striped mantine-Table-striped __mantine-ref-hover mantine-Table-hover mantine-1fw2ryq"
    >
      <caption>
        Hola mundo
      </caption>
      <thead>
        <tr />
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`CustomTable tests should match style 1`] = `
<div>
  <div
    class="sc-bczRLJ LwpKF"
  >
    Table
  </div>
</div>
`;
